test/foundry/invariant/handlers/AccountCreationHandlerTest.t.sol

```diff
@@ -0,0 +1,127 @@
+        bytes memory initData = abi.encodePacked(owner);
+        address payable newAccount = accountFactory.createAccount(validationModule, initData, index);
+        require(newAccount != address(0), "Account creation failed");
+        lastCreatedAccount = newAccount; // Update the last created account
+
+        // Validate if the account is a contract and correctly set up
+        require(isContract(newAccount), "Created account is not a valid contract");
+
+        // Initial balance should be zero
+        assertAccountBalance(newAccount, 0 ether);
+
+        // Initial nonce should be zero
+        assertNonce(newAccount, nonceKey, 0);
+
+        // Validate account creation
+        assertValidCreation(Nexus(newAccount));
+    }
+
+    /// @notice Ensures nonce consistency before and after account creation.
+    /// @param index The index for account creation.
+    function invariant_nonceConsistency(uint256 index) external {
+        address payable newAccount = accountFactory.createAccount(validationModule, abi.encodePacked(owner), index);
+        assertTrue(newAccount != address(0), "Account creation failed");
+
+        // Calculate the expected nonce based on the newly created account and the validation module
+        uint256 expectedNonce = getNonce(newAccount, validationModule);
+        uint256 actualNonce = ENTRYPOINT.getNonce(newAccount, uint192(bytes24(bytes20(address(validationModule)))));
+
+        // Assert that the calculated nonce matches the actual nonce
+        assertEq(actualNonce, expectedNonce, "Nonce consistency invariant violated after account creation");
+        assertValidCreation(Nexus(newAccount));
+    }
+
+    /// @notice Verifies that the nonce is reset to zero upon account creation.
+    /// @param index The index for account creation.
+    function invariant_nonceResetOnCreation(uint256 index) external {
+        address payable newAccount = accountFactory.createAccount(validationModule, abi.encodePacked(owner), index);
+        assertTrue(newAccount != address(0), "Account creation failed");
+
+        // Retrieve the nonce for the newly created account
+        uint256 nonceAfterCreation = getNonce(newAccount, validationModule);
+
+        // Assert that the nonce is zero for a new account
+        assertEq(nonceAfterCreation, 0, "Nonce should be reset to zero upon account creation");
+        assertValidCreation(Nexus(newAccount));
+    }
+
+    /// @notice Tests the creation of multiple accounts with different indices and validates nonce initialization.
+    function invariant_multipleAccountCreationWithUniqueIndices() external {
+        uint256 index1 = 1;
+        uint256 index2 = 2;
+
+        address payable account1 = accountFactory.createAccount(validationModule, abi.encodePacked(owner), index1);
+        address payable account2 = accountFactory.createAccount(validationModule, abi.encodePacked(owner), index2);
```

Comment:
all of these will start breaking post factory changes

Reply:
🥶